Algorithms-Merge-Sort.md

```diff
@@ -0,0 +1,53 @@
+# Merge Sort
+
+Most modern languages have an inbuilt sort() function which automatically sorts an input array or list. Did you ever wonder how the sort function actually works on the inside?. Knowing common sorting algorithms and their implementations is the most important part of a coding interview. In this series of articles, we will look at several important sorting algorithms. How they are implemented, the time and space complexity etc. Our very first post is on Merge Sort.
+
+To learn about Merge Sort, a basic knowledge about Recursion is a pre-requisite. Merge Sort is based on the principle of Divide and Conquer. The whole process of sorting an array of N integers can be summarized into three steps-
+
+- Divide the array into two halves.
+- Sort the left half and the right half using the same recurring algorithm.
+- Merge the sorted halves.
+
+The biggest advantage of using Merge sort is that it takes only n*log(n) steps to sort an entire Array. It is a lot better than n^2 running time of bubble sort or insertion sort.
```

Comment:
We CS students may instantly grasp the concept of complexity `n*log(n)`, but to someone else it might be confusing about what this term means. Either you can add a short intro, or you can put appropriate link to external resources.